python/mxnet/optimizer.py

```diff
@@ -650,6 +650,47 @@ def update(self, index, weight, grad, state):
         weight[:] -= current_delta + wd * weight
 
 @register
+class Ftrl(Optimizer):
+    """
+    Reference:Ad Click Prediction: a View from the Trenches
+    """
+
+    def __init__(self, lamda1=0.01, alpha=0.1, beta=1, **kwargs):
+        super(Ftrl, self).__init__(**kwargs)
+        self.lamda1 = lamda1
+        self.alpha = alpha
+        self.beta = beta
+
+
+    def create_state(self, index, weight):
+        return (zeros(weight.shape, weight.context),  # sigma
```

Comment:
I find that sigma is not necessary since it's always related to n. I think we can use
```python
z += grad - (sqrt(n + grad * grad) - sqrt(n)) / self.alpha * weight
n += grad * grad
```